modules/electrophysiology_browser/php/models/electrophysioevents.class.inc

```diff
@@ -57,7 +78,340 @@ class ElectrophysioEvents
+
+            if (is_null($instance_id)) {
+                // TODO: Support Instance INSERT
+                $instance_id = 1;
+            }
+
+            $instance = [
+                'Onset'    => $instance_data['onset'],
+                'Duration' => $instance_data['duration'],
+            ];
+
+            // TODO: Support Event Instance Insert
+            if (!empty($eventFileID)) {
+                // Update physiological_task_event
+                $db->update(
+                    'physiological_task_event',
+                    $instance,
+                    ['PhysiologicalTaskEventID' => $instance_id]
+                );
+
+                $db->update(
+                    'physiological_event_file',
+                    ['LastUpdate' => date("Y-m-d H:i:s")],
+                    ['PhysiologicalFileID' => $this->_physioFileID]
+                );
+            }
+
+            $taskEvent = $db->pselect(
+                'SELECT * FROM physiological_task_event
+            WHERE PhysiologicalTaskEventID=:PTEID',
+                ['PTEID' => $instance_id]
+            );
+
+            $extraColumns = $db->pselect(
+                'SELECT opt.*
+                  FROM physiological_task_event_opt AS opt
+                  WHERE opt.PhysiologicalTaskEventID=:PTEID',
+                ['PTEID' => $instance_id]
+            );
+
+            return [
+                'instance'      => $taskEvent[0],
+                'extra_columns' => $extraColumns,
+            ];
+        }
+        return [];
+    }
+
+    /**
+     * Deletes one event instance
+     *
+     * @param int $physiologicalTaskEventID PhysiologicalTaskEventID
+     *
+     * @return void
+     */
+    function deleteEvent(int $physiologicalTaskEventID): void
+    {
+        $db = \NDB_Factory::singleton()->database();
+
+        $physioFileID = $db->pselectone(
+            'SELECT PhysiologicalFileID
+              FROM physiological_task_event
+              WHERE PhysiologicalTaskEventID=:taskEventID',
+            ['taskEventID' => $physiologicalTaskEventID]
+        );
+
+        // TODO: Check that this cascades properly to rel tables
+        if ($this->_physioFileID == $physioFileID) {
+            $db->delete(
+                "physiological_task_event",
+                ['PhysiologicalTaskEventID' => $physiologicalTaskEventID]
+            );
+        }
+    }
+
+    /**
+     * Updates the event files associated with the given
+     * physiological file ID
+     *
+     * @return void
+     * @throws SodiumException
+     */
+    function updateFiles(): void
+    {
+        $db = \NDB_Factory::singleton()->database();
+
+        //Get data directory base path from Config
+        $dataDir = $db->pselectOne(
+            'SELECT Value
+            FROM Config AS config
+            INNER JOIN ConfigSettings AS c
+            ON c.Name=:name AND config.ConfigID=c.ID',
+            ['name' => 'dataDirBasepath']
+        );
+
+        $tsv = $db->pselect(
```

Comment:
As a note, not sure if it is the intended behaviour, use `pselectRow` instead to remove the need for `$tsv[0]` after?

Reply:
This was copied verbatim from `electrophysioannotations.class.inc`